docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,562 @@
+# Diamond workflow support
+
+A diamond workflow allows for a pipeline to scale out the generation of a single bundle.
+
+A diamond is composed of multiple splits; each split tries to write a set of files.
+Ideally, 2 splits do not write the same file.
+
+To elaborate, a bundle consists of a filesystem namespace with files in folders; each split can write files to any
+folder but should not try to write the same file.
+
+The individual nodes add files to splits without any co-ordination between them.
```

Comment:
more commentary for clarity rather than fundamental idea:

could this be clarified as

> The individual nodes _in the DAG generated by the Argo runtime_ add files to splits without any co-ordination between them.

otherwise, it's not totally clear to me what vertex set is referred to here

Reply:
I am just mentioning "nodes" here. There is no reason to introduce DAG, ARGO, vertices or other complex notions. A node simply executes an instance of the datamon program.